app/validators/link_or_identifier_validator.rb

```diff
@@ -0,0 +1,37 @@
+class LinkOrIdentifierValidator < ActiveModel::Validator
+  def validate(record)
+    if link_and_identifier_exists?(record) || link_and_identifier_missing?(record)
```

Comment:
Could there be a better name for these functions? Because it reads like: trigger an error if either link and id exists, or link and id are missing.

Reply:
Sure, we can rejig the naming if it's not clear, however your assumption of 'trigger an error if either link and id exists, or link and id are missing.'  is correct. We only want to trigger an alert if:

a) the link AND identifier fields are both filled in or 
b) the link AND identifier fields are both blank. 

A user filling in one OR the other is the correct procedure.

How about, `both_link_and_identifier_exist? || both_link_and_identifier_missing?` - thoughts? Happy for other suggestions too!